### PR TITLE
osdocs-82 Add how builds work assembly

### PR DESCRIPTION
**Pull request by bmcelvee**
https://jira.coreos.com/browse/OSDOCS-82

### CODE REVIEW

File: builds/understanding-builds.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+toc::[]
+
+The following sections define the concept of a build, and outline the primary
```
Comment (bmcelvee) [inline builds/understanding-builds.adoc]: Distinguish that this section is OpenShift specific (not Kube as deployments are)

**Comment by bmcelvee** [comment]
@openshift/team-documentation PTAL :snowflake:

**Comment by bmcelvee** [comment]
FYI - I'm adding the build strategy modules in https://github.com/openshift/openshift-docs/pull/13000

File: modules/builds-buildconfig.adoc [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+endif::[]
+or `Binary`, to accept binary payloads. It is possible to have multiple
+sources at once, refer to the documentation for each source type for details.
+<5> The `strategy` section describes the build strategy used to execute the
+build. You can specify a `Source`
+ifndef::openshift-online[]
+, `Docker`, or `Custom`
+endif::[]
+strategy here. This above example uses the `ruby-20-centos7` container image
+that Source-To-Image will use for the application build.
```
Comment (kalexand-rh) [inline modules/builds-buildconfig.adoc]: s/will use/uses

File: modules/builds-buildconfig.adoc [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+A build configuration describes a single build definition and a set of
+triggers for when a new build should be created. Build configurations are defined
+by a `BuildConfig`, which is a REST object that can be used in a POST to the API
+server to create a new instance.
+
+Depending on how you choose to create your application using {product-title}, a
+`BuildConfig` is typically generated automatically for you if you use the web
+console or CLI, and it can be edited at any time. Understanding the parts that
+make up a `BuildConfig` and their available options can help if you choose to
+manually tweak your configuration later.
```
Comment (kalexand-rh) [inline modules/builds-buildconfig.adoc]: I'd s/tweak/change

File: modules/builds-about.adoc [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+[id='builds-about-{context}']
+= What is a build?
+
+A _build_ is the process of transforming input parameters into a resulting
+object (build object). Most often, the process is used to transform input
+parameters or source code into a runnable container image.
+
+A _build configuration_, or `BuildConfig`, is characterized by a _build strategy_
+and one or more sources. The strategy determines the aforementioned process, while
```
Comment (kalexand-rh) [inline modules/builds-about.adoc]: I'd remove "aforementioned"

File: modules/builds-about.adoc [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+the sources provide its input.
+
+{product-title} leverages Kubernetes by creating containers
```
Comment (kalexand-rh) [inline modules/builds-about.adoc]: s/leverages/uses

File: modules/builds-about.adoc [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+objects share common characteristics including inputs for a build, need to
+complete a build process, logging the build process, and publishing resources from
+successful builds and the final status of the build.
+
+The {product-title} build system provides extensible support for _build
+strategies_ that are based on selectable types specified in the build API.
+
+*The build strategies are:*
+
+* Docker build
```
Comment (kalexand-rh) [inline modules/builds-about.adoc]: Is it still "Docker?"

File: modules/builds-about.adoc [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+* Binary (Local) inputs
+* Input secrets
+* External artifacts
+
+The resulting object of a build depends on the builder used to create it. For
+Docker and S2I builds, the resulting objects are runnable images. For Custom
+builds, the resulting objects are whatever the builder image author has
+specified.
+
+Additionally, the Pipeline build strategy can be used to implement sophisticated
```
Comment (kalexand-rh) [inline modules/builds-about.adoc]: Pipeline isn't in the list of available build strategies.

File: modules/builds-about.adoc [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+*The build strategies are:*
+
+* Docker build
+* Source-to-Image (S2I) build
+* Custom build
+
+*The six types of sources that can be given as _build input_ are:*
+
+ifndef::openshift-online[]
+* Inline Dockerfile definitions
```
Comment (kalexand-rh) [inline modules/builds-about.adoc]: If "Dockerfile" is only in online, should Docker be listed as an available build strat for all versions?

**Review by kalexand-rh — COMMENTED** [review]
Couple of picks and some modularization comments. Otherwise LGTM!

File: modules/builds-about.adoc (hunk 6 above)
Comment (bmcelvee) [reply modules/builds-about.adoc]: Yep! I asked about this yesterday.

File: modules/builds-about.adoc (hunk 7 above)
Comment (bmcelvee) [reply modules/builds-about.adoc]: Added :)

File: modules/builds-about.adoc (hunk 8 above)
Comment (bmcelvee) [reply modules/builds-about.adoc]: It's in everything but (was `ifndef` changed to `ifdef`)

**Pull request merged**